Merge the work on lung fibrosis with main

eioannoucy opened the pull request:
As the work has been completed and publish we can merge with main.

eioannoucy committed:
Bug fix in ripf parameters

eioannoucy committed:
Simplification of tumour/fibrosis volume calculation

eioannoucy committed:
Allow for specifying input file. If not, it will use input.dat for backwards compatibility.

Add ReadMe with instructions on how to build.

eioannoucy committed:
Model changes

1. Tau has exponentianl only on total volume fraction
2. Lambda prodution does not have a HU term
3. Omicro production has a HU term

Some renaming of a few parameters for simplicity

eioannoucy committed:
Model correction. Add Tau derivative with respect to HU

Removed explicit HU dependance for omecro production term. It is included in Tau

eioannoucy committed:
Model update. Change in growth production term.

eioannoucy committed:
Check for overshoots

eioannoucy committed:
Model change on fibrosis initation term and exposure to settings file

eioannoucy committed:
ripf code refactoring (stage 1)

eioannoucy committed:
ripf code refactoring (stage 2)

eioannoucy committed:
ripf code refactoring (stage 3)

eioannoucy committed:
ripf code refactoring (stage 4)

eioannoucy committed:
ripf code refactoring (stage 5)

eioannoucy committed:
Change in HU equation

Restrict cc proliferation for low values. It was observed to lead to cc growth at the edge of the lung

eioannoucy committed:
Remove Pc expression from cc equation

eioannoucy committed:
Change in capacity term to allow for higher HU

eioannoucy committed:
Fix for fb_prol_dfb when fb=0 (division with zero)

eioannoucy committed:
Detect and ignore negative fb rate for ecm equation

eioannoucy committed:
Bug fix in settings file output

eioannoucy committed:
Changes to align with main

eioannoucy committed:
Merge branch 'main' into lefteris-ripf

eioannoucy committed:
Restore use of exodus (ex2) for output after aligining with main

vasvav commented:
The following amendments need to be considered:
- [x] move the *input.dat.example* into the corresponding examples folder by replacing the existing input file *rdcFEs/run/RIPF133/input.dat*
- [x] make sure the updated input file makes use of the corresponding <ins>dat</ins> and <ins>mash</ins> files available in the *rdcFEs/run/RIPF133/* directory
- [x] replace all tabs in the source code files *rdcFEs/src/ripf.C* and *rdcFEs/src/main.C* with white spaces, and confirm code has been properly indented
- [x] revise code in lines 38-45 of *main.C* such that the model case "ripf" is coded as with the ones above
- [x] doublecheck the if-statement in lines 446-450 of *ripf.C* is correctly positioned! maybe it should have been located outside the scope (after) the for-loop (lines 438-451)
- [x] what is the purpose of the if-statement in lines 684-687 of *ripf.C*, if not necessary then remove it!
- [x] can you doublecheck whether the if-statement in line 503 of *ripf.C* is valid having (considering the one in which it's nested, i.e., see line 497 where you have `if (fb_old >= 0.0 && fb_old < 1.0)`

eioannoucy committed:
Update example in run/RIPF133

Remove tabs from src files

Align treatment of input file of ripf with the other modules

eioannoucy committed:
Move negative fb derivative check, as per Vasilis's suggestion

eioannoucy commented:
I believe I addressed the comments in the pushed changes.

On the if-statements:

- The if-statement at l. 441 is a check for negative fb derivative which shouldn't happen in any case. I have moved it out of the loop but I think it works either way as it can check during the loop or at the end of it and it should still catch the error.
- The if-statement at l. 678 checks if fb has overshoot over 1, the maximum value of the variable. This happened for a few patients when the rate is stiff and requires reduction of the time step.
- The if-statement at l.497 is so that we avoid a big calculation if the fb (and the expression) value is zero, which is the case in most of the domain. The previous check allows` fb >= 0` (also equal to zero) but the later expression is computed only when `fb > 0`

vasvav reviewed (approved):
Great job @eioannoucy... PR approved!
